implementations/micrometer-registry-timestream2/src/main/java/io/micrometer/timestream2/TimestreamMeterRegistry.java

```diff
@@ -0,0 +1,333 @@
+import static java.util.stream.StreamSupport.stream;
+import static software.amazon.awssdk.services.timestreamwrite.model.TimeUnit.SECONDS;
+
+/**
+ * {@link MeterRegistry} for TimeStream.
+ *
+ * @author Guillaume Hiron
+ * @since 1.6.0
+ */
+public class TimestreamMeterRegistry extends StepMeterRegistry {
+
+    static final ThreadFactory DEFAULT_THREAD_FACTORY = new NamedThreadFactory("timestream-metrics-publisher");
+
+    private static final WarnThenDebugLogger warnThenDebugLogger = new WarnThenDebugLogger(TimestreamMeterRegistry.class);
+
+    private final Logger logger = LoggerFactory.getLogger(TimestreamMeterRegistry.class);
+
+    private final TimestreamConfig config;
+
+    private final TimestreamWriteAsyncClient timestreamWriteAsyncClient;
+
+    /**
+     * Create a new instance with given parameters.
+     *
+     * @param config        configuration to use
+     * @param clock         clock to use
+     * @param threadFactory thread factory to use
+     * @param client        TimeStream client to use
+     */
+    //VisibleForTesting
+    TimestreamMeterRegistry(TimestreamConfig config, Clock clock, ThreadFactory threadFactory, TimestreamWriteAsyncClient client) {
+        super(config, clock);
+        config().namingConvention(new TimestreamNamingConvention());
+        this.config = config;
+        checkClient(client);
+        this.timestreamWriteAsyncClient = client;
+        start(threadFactory);
+    }
+
+    private void checkClient(TimestreamWriteAsyncClient client) {
+        Objects.requireNonNull(client);
+    }
+
+    @Override
+    protected void publish() {
+        MeterPartition.partition(this, config.batchSize())
+                .stream()
+                .map(this::toRecords)
+                .forEach(this::sendRecords);
+    }
+
+    //VisibleForTesting
+    List<Record> toRecords(List<Meter> partition) {
+        Batch batch = new Batch();
+        return partition.stream().flatMap(m -> m.match(
+                batch::gaugeData,
+                batch::counterData,
+                batch::timerData,
+                batch::summaryData,
+                batch::longTaskTimerData,
+                batch::timeGaugeData,
+                batch::functionCounterData,
+                batch::functionTimerData,
+                batch::metricData)
+        ).collect(toList());
+    }
+
+    //VisibleForTesting
+    void sendRecords(List<Record> records) {
+        WriteRecordsRequest writeRecordsRequest = WriteRecordsRequest.builder()
+                .databaseName(config.databaseName())
+                .tableName(config.tableName())
+                .records(records)
+                .build();
+
+        timestreamWriteAsyncClient
+                .writeRecords(writeRecordsRequest)
+                .thenRun(() ->
+                        logger.debug("published records to database: {}, table: {} ", writeRecordsRequest.databaseName(), writeRecordsRequest.tableName())
+                )
```

Comment:
Is this too long to just make this one line?
```suggestion
                .thenRun(() -> logger.debug("published records to database: {}, table: {} ", writeRecordsRequest.databaseName(), writeRecordsRequest.tableName()))
```